### ISSUE TITLE
Any documentation about becoming dependency free?

### ISSUE THREAD

**Issue by grug** [graphql/graphql-js#2380]
Hey there - big fan of the library and a big fan of the fact there are zero dependencies in the project. I was just wondering if there is any documentation/blog posts etc... I can read about the strategies taken to remove all the dependencies from the project? I'm just really interested in the strategies and lessons learned from this effort that I could apply to my own projects :)

Thanks
-Dave

**Comment by IvanGoncharov** [comment]
@grug Thanks for the interest 👍 
Actually, This library was dependency-free until `iterall` dependency was introduced almost 4 years ago:
https://github.com/graphql/graphql-js/commit/03f06e9a901df62d7a0df19860db3cd5106b0c3f#diff-b9cfc7f2cdf78a7f4b91a753d10865a2
So it's not a very exciting story: I just removed `iterall` that's it 🤷‍♂️

In general, graphql-js is a reference implementation that means it should be easily portable to other languages so we can't depend on any 3rd-party library.
`iterall` was special in a sense it's more a polyfill than a library but since it started to cause issues with ESM build I just removed it.

Did I answer your question?
